src/clj/puppetlabs/services/request_handler/request_handler_core.clj

```diff
@@ -259,21 +260,35 @@
     (= (:type x)
        :puppetlabs.services.request-handler.request-handler-core/bad-request)))
 
+(defn jruby-timeout?
+  [x]
+  "Determine if the supplied slingshot message is for a JRuby borrow timeout."
+  (when (map? x)
+    (= (:type x)
+       ::jruby/jruby-timeout)))
```

Comment:
So the `::` was initially a typo which, for some very mysterious reason, causes the test to pass. When I changed it to `:jruby/jruby-timeout` it failed, as expected. I'll change this to `:puppetlabs.services.jruby.jruby-puppet-service/jruby-timeout` to clear up the ambiguity. But this is super strange. Anyone have any ideas?